daemon.go

```diff
@@ -384,34 +461,7 @@ func main() {
 
 	defer watcher.Close()
 
-	if *flag_recursive == true {
-		err = filepath.Walk(*flag_directory, func(path string, info os.FileInfo, err error) error {
-			if err == nil && info.IsDir() {
-				if flag_excludedDirs.Matches(path) {
-					return filepath.SkipDir
-				} else {
-					if *flag_verbose {
-						log.Printf("Watching directory '%s' for changes.\n", path)
-					}
-					return watcher.Add(path)
-				}
-			}
-			return err
-		})
-
-		if err != nil {
-			log.Fatal("filepath.Walk():", err)
-		}
-
-		if err := watcher.Add(*flag_directory); err != nil {
-			log.Fatal("watcher.Add():", err)
-		}
-
-	} else {
-		if err := watcher.Add(*flag_directory); err != nil {
-			log.Fatal("watcher.Add():", err)
-		}
-	}
+	watchDirectories(watcher)
```

Comment:
I think this is not the best way to refactor this. How about this:
```go
func watchDirectory(dir string, watcher *fsnotify.Watcher) {
// ...
}

func main() {
// ...
    for dir := range flag_directories { 
        watchDirectory(dir, watcher)
    }
}

Reply:
I structured it this way (along with `validateFlags`)  with an eye on cleaning up the `main` function since it's doing a lot directly (many things could stand to be broken out into more functions).  However, it's not that big of a deal so I'm happy to make the change.